docs/plugins.md

```diff
@@ -0,0 +1,352 @@
+
+**Type**: `Array<{path: string}>`
+
+
+#### `category`
+
+Defines the display strings of the plugin's category and configures audit scoring and grouping. It is an object with at least two properties `title` and `auditRefs`.
+
+- `title: string` **REQUIRED** - The display name of the plugin's category in the report.
+- `description: string` *OPTIONAL* - A more detailed description of the category's purpose.
+- `manualDescription: string` *OPTIONAL* - A more detailed description of all of the manual audits in a plugin. Only use this if you've added manual audits.
+- `auditRefs: Array<{id: string, weight: number, group?: string}>` **REQUIRED** - The list of audits to include in the plugin category along with their overall weight in the score of the plugin category. Each audit ref may optionally reference a group ID from `groups`.
+
+
+#### `groups`
+
+Defines the audit groups used for display in the HTML report.
+
+**Example of Audit Display with Groups**
+![audit group with groups](https://user-images.githubusercontent.com/2301202/56936017-86d3ce80-6aba-11e9-9a43-39bf3810b551.png)
+
+**Example of Audit Display without Groups**
+![audit display without groups](https://user-images.githubusercontent.com/2301202/56936043-c0a4d500-6aba-11e9-9e37-0bc131010a37.png)
+
+It is an object whose keys are the group IDs and whose values are objects with the following properties:
+
+- `title: string` **REQUIRED** - The display name of the group in the report.
+- `description: string` *OPTIONAL* - A more detailed description of the group's purpose.
+
+### Plugin Audits
+
+A plugin audit is a class that implements at least two properties: `meta` and `audit()`.
+
+#### `meta`
+
+The `meta` property is a static getter for the metadata of an audit. It should return an object with the following properties:
+
+- `id: string` **REQUIRED** - The string identifier of the audit, in kebab case, typically matching the file name.
+- `title: string` **REQUIRED** - Short, user-visible title for the audit when successful.
+- `failureTitle: string` *OPTIONAL* - Short, user-visible title for the audit when failing.
+- `description: string` **REQUIRED** - A more detailed description that describes why the audit is important and links to Lighthouse documentation on the audit; markdown links supported.
+- `requiredArtifacts: Array<string>` **REQUIRED** - A list of artifacts that must be present for the audit to execute. See [Available Artifacts](#available-artifacts) for what's available to plugins.
+- `scoreDisplayMode: "numeric" | "binary" | "manual" | "informative"` *OPTIONAL* - A string identifying how the score should be interpreted for display.
+
+See [Best Practices > Naming](#naming) for best practices on the display strings.
+
+#### `audit(artifacts, context)`
```

Comment:
i think we could also link up adspeed insights plugin as an example of a complex but well written plugin

Reply:
good idea!